src/test/groovy/net/vivin/gradle/versioning/TestRepository.groovy

```diff
@@ -130,4 +135,15 @@ class TestRepository {
             .setTarget(repository.resolve(Constants.HEAD))
             .call()
     }
+
+    String getHeadTagMessage() {
+        def revWalk = new RevWalk(repository)
+        def head = repository.resolve(Constants.HEAD)
+
+        return repository.tags.values()
+            .collect { repository.peel(it) }
+            .findAll { it.peeledObjectId == head }
+            .collect { revWalk.parseTag(it.objectId).fullMessage }
+            .find()
+    }
```

Comment:
I hope you didn't have a reason to make it that complex.
I exchanged this with a much simpler method, have a look. :-)